### PR TITLE
ensure-kustomize

### PR DESCRIPTION
**Pull request by 2uasimojo**
Add a convenience target for consumers to ensure the `kustomize` utility is installed.

### CODE REVIEW

**Comment by 2uasimojo** [comment]
/assign @soltysh 

@soltysh This one got weird -- and I think it uncovered a bug in the way includes are being handled. How does this repo handle bugs? Issue, BZ, Jira, slack, IRC, carrier pigeon....?

**Comment by soltysh** [comment]
> @soltysh This one got weird -- and I think it uncovered a bug in the way includes are being handled. How does this repo handle bugs? Issue, BZ, Jira, slack, IRC, carrier pigeon....?

GH issue or just slack should work fine. 

/hold

Why and where do we need kustomize, is another question I'd like to get answered before merging this in.

**Comment by 2uasimojo** [comment]
> GH issue or just slack should work fine.

Rockin. #43. Happy to propose that fix if you agree with the suggested approach.

> Why and where do we need kustomize, is another question I'd like to get answered before merging this in.

Yah, fair. I'm coming from [this hive doc](https://github.com/openshift/hive/blob/master/docs/developing.md#external-tools). Hive [uses kustomize](https://github.com/openshift/hive/blob/1da8686c66d6d97e0498b4596ad4b9040ae8073d/Makefile#L184-L185) to assemble and edit yaml manifests for k8s objects. I believe this is a very common pattern, including among projects using [operator-sdk](https://sdk.operatorframework.io/docs/cli/operator-sdk_generate_kustomize_manifests/).

File: make/targets/openshift/kustomize.mk [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+KUSTOMIZE_VERSION ?= 4.1.3
+KUSTOMIZE ?= $(PERMANENT_TMP_GOPATH)/bin/kustomize
+kustomize_dir := $(dir $(KUSTOMIZE))
+
+ensure-kustomize:
+ifeq "" "$(wildcard $(KUSTOMIZE))"
+	$(info Installing kustomize into '$(KUSTOMIZE)')
+	mkdir -p '$(kustomize_dir)'
+	@# NOTE: Pinning script to a tag rather than `master` for security reasons
+	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/kustomize/v$(KUSTOMIZE_VERSION)/hack/install_kustomize.sh"  | bash -s $(KUSTOMIZE_VERSIO) $(kustomize_dir)
```
Comment (2uasimojo) [inline make/targets/openshift/kustomize.mk:18]: oops
```suggestion
	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/kustomize/v$(KUSTOMIZE_VERSION)/hack/install_kustomize.sh"  | bash -s $(KUSTOMIZE_VERSION) $(kustomize_dir)
```

File: make/targets/openshift/kustomize.mk [hunk 2]
```diff
@@ -0,0 +1,30 @@
+self_dir_kustomize :=$(dir $(lastword $(MAKEFILE_LIST)))
+# We need to include this before we use PERMANENT_TMP_GOPATH
+# (indirectly) from ifeq. We use a unique variable name to store the pwd
+# so as not to conflict with `self_dir` elsewhere.
+include $(addprefix $(self_dir_kustomize), \
+	../../lib/tmp.mk \
+)
```
Comment (2uasimojo) [inline make/targets/openshift/kustomize.mk:7]: See #43.
```suggestion
# We need to include this before we use PERMANENT_TMP_GOPATH
# (indirectly) from ifeq.
include $(addprefix $(dir $(lastword $(MAKEFILE_LIST))), \
	../../lib/tmp.mk \
)
```

**Review by 2uasimojo — COMMENTED** [review]
I'll wait for @soltysh's review before fixing these couple of things.

**Review by soltysh — COMMENTED** [review]
I just tagged #43, can you fix the nits you've pointed out in here and I'll tag. 
Overall this lgtm.

**Comment by 2uasimojo** [comment]
> I just tagged #43, can you fix the nits you've pointed out in here and I'll tag.
> Overall this lgtm.

Thanks @soltysh! Done and squashed.

**Comment by 2uasimojo** [comment]
@soltysh Can we remove the `hold` now?

**Pull request merged**